From 9adec3ff5c2ca17855fd67c537a4025c36a96f77 Mon Sep 17 00:00:00 2001
From: Volodymyr Bendiuga <volodymyr.bendiuga@westermo.se>
Date: Fri, 20 Oct 2017 13:09:49 +0200
Subject: [PATCH 4/5] tc:nat: convert mask to host byte order before showing to
 user
Organization: Westermo Network Technologies AB

Otherwise on little endian machines it will show up incorrectly
and will make it very confusing.

Signed-off-by: Volodymyr Bendiuga <volodymyr.bendiuga@westermo.se>
Signed-off-by: Joachim Nilsson <troglobit@gmail.com>
---
 tc/m_nat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tc/m_nat.c b/tc/m_nat.c
index c4b02a83..5a551d05 100644
--- a/tc/m_nat.c
+++ b/tc/m_nat.c
@@ -157,7 +157,7 @@ print_nat(struct action_util *au, FILE * f, struct rtattr *arg)
 	}
 	sel = RTA_DATA(tb[TCA_NAT_PARMS]);
 
-	len = ffs(sel->mask);
+	len = ffs(ntohl(sel->mask));
 	len = len ? 33 - len : 0;
 
 	print_string(PRINT_ANY, "type", " %s ", "nat");
-- 
2.25.1

